src/libraries/Microsoft.Extensions.DependencyInjection/tests/DI.Tests/ServiceProviderContainerTests.cs

```diff
@@ -410,6 +410,89 @@ public void Dispose()
         [ThreadStatic]
         public static int ThreadId;
 
+        private class Foo
```

Comment:
Can we name these classes a little more descriptive than `Foo` and `Bar`? Maybe `OuterSingleton` and `InnerSingleton`? I'm open to other names too - those just came to the top of my head.